### DISCUSSION TITLE
Cannot measure vulnerability while disable OSS index and enable internal analyzer

### DISCUSSION THREAD

**Discussion by ninianxing** [DependencyTrack/dependency-track#1955] [Q&A]
Enable internal analyzer only cannot measure vulnerability of component. I can't see severity and scores of components, only if I enable the OSS index.
![image](https://user-images.githubusercontent.com/103549973/190111781-fd72c529-47bd-4812-af3d-ae85e309b7d4.png)
![image](https://user-images.githubusercontent.com/103549973/190111890-e8c2d22b-5414-48ca-a7af-50447dcce6f6.png)

**Comment by syalioune** [comment]
Hello @ninianxing,
Can you please provide the spec (i.e. package url and/or CPE identifier) of the incriminated component ?
It is most likely that your component only contains packageurl and thus cannot be checked against NVD database which only use CPE identifier.

**Reply by ninianxing** [reply]
Hi,
The SBOM contains no CPE info, so that there is no CPE in component. I found that if use OSS index analyzer in Internet, vulnerability would be measured without CPE. But if use internal analyzer in private network, it wouldn't be measured. 
How can I measure it without CPE info ? Or filling CPE info in component automatically?

![微信截图_20220915104134](https://user-images.githubusercontent.com/103549973/190301142-3330907a-09b0-407c-b991-284e0688226a.png)

**Comment by syalioune** [comment]
> How can I measure it without CPE info ?

It all boils down to what vulnerability intelligence datasource you are using :

**Internal Analyzer :** It is actually the addition of many vuln intelligence source
- NVD : Fetched from internet but can be mirrored in intranet private network - Only works with CPE
- Github Advisories : Fetched from internet but cannot at the moment be mirrored - Only works with PURL
- Google OSV (coming in 4.6) : Fetched from internet but cannot at the moment be mirrored - Only works with PURL

**OSS Index Analyzer :** Fetched from internet but cannot at the moment be mirrored - Only works with PURL

**VulnDB (need a license) :** Two modes
- VulnDB REST API : On the fly analysis fetched from internet but cannot be mirrored - Only work with CPE
- VulnDB data mirror : Data feed on disk which is sync by DT. Can be offline - Only work with CPE

> Or filling CPE info in component automatically?

Guessing CPE from other data/identifier like purl is tricky business. I would not know where to even start !
